x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/ExtractedFieldsDetector.java

```diff
@@ -304,22 +368,42 @@ private ExtractedFields detectExtractedFields(Set<String> fields, Set<FieldSelec
         return extractedFields;
     }
 
-    private ExtractedFields deduplicateMultiFields(ExtractedFields extractedFields, boolean preferSource,
+    private ExtractedFields deduplicateMultiFields(ExtractedFields extractedFields,
+                                                   boolean preferSource,
                                                    Set<FieldSelection> fieldSelection) {
-        Set<String> requiredFields = config.getAnalysis().getRequiredFields().stream().map(RequiredField::getName)
+        Set<String> requiredFields = config.getAnalysis()
+            .getRequiredFields()
+            .stream()
+            .map(RequiredField::getName)
             .collect(Collectors.toSet());
         Map<String, ExtractedField> nameOrParentToField = new LinkedHashMap<>();
         for (ExtractedField currentField : extractedFields.getAllFields()) {
+            // If this field or its parent is processed ONLY, then its fine. We don't need to deduplicate it as it won't be sent
+            // directly to the native process
+            if (extractedFields.getProcessedOnlyFields().contains(currentField.getName())) {
+                nameOrParentToField.put(currentField.getName(), currentField);
+                continue;
+            }
+            if (currentField.isMultiField() && extractedFields.getProcessedOnlyFields().contains(currentField.getParentField())) {
+                nameOrParentToField.put(currentField.getParentField(), currentField);
+                continue;
+            }
             String nameOrParent = currentField.isMultiField() ? currentField.getParentField() : currentField.getName();
             ExtractedField existingField = nameOrParentToField.putIfAbsent(nameOrParent, currentField);
+            // TODO, there is an issue where the processed field references the field `foo.keyword` but an included field references `foo`
+            // The included field might only be `foo`. How should we adjust the `foo.keyword` inclusion?
+            // It seems to me that we should disallow users from asking for a `foo.keyword` field in the processors.
```

Comment:
@dimitris-athanasiou I am torn with what to do here.

Reply:
Well, if we prevent folks from including a "organically" AND via a feature_processor, we could skirt this issue.

Reply:
I am going to implement the following restriction:

- If the field is referenced in a `feature_processor` it cannot be included directly in analytics. 
- We will provide an `identity` processor so that folks can do this themselves in the future if necessary.